relay-dynamic-config/src/global.rs

```diff
@@ -381,7 +381,11 @@ fn is_ok_and_empty(value: &ErrorBoundary<MetricExtractionGroups>) -> bool {
 fn is_missing(value: &ErrorBoundary<ModelCosts>) -> bool {
     matches!(
         value,
-        &ErrorBoundary::Ok(ModelCosts{ version, ref costs }) if version == 0 && costs.is_empty()
+        &ErrorBoundary::Ok(ModelCosts {
+            ref models,
+            ref costs,
+            version,
+        }) if (models.is_empty() && version == 2) || (costs.is_empty() && version == 1)
```

Comment:
let's do what `metric_extraction` does, we can add an `is_empty()` to `ModelCosts` and just use that and possibly drop the version checks as they are kind of pointless (as we discussed).